Common/Brokerages/TDameritradeBrokerageModel.cs

```diff
@@ -0,0 +1,120 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using QuantConnect.Orders;
+using QuantConnect.Orders.Fees;
+using QuantConnect.Securities;
+using QuantConnect.Util;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.Brokerages
+{
+    /// <summary>
+    /// TDAmeritrade
+    /// </summary>
+    public class TDAmeritradeBrokerageModel : DefaultBrokerageModel
+    {
+        /// <summary>
+        /// Gets a map of the default markets to be used for each security type
+        /// </summary>
+        public override IReadOnlyDictionary<SecurityType, string> DefaultMarkets { get; } = GetDefaultMarkets();
+
+        /// <summary>
+        /// Constructor for TDAmeritrade brokerage model
+        /// </summary>
+        /// <param name="accountType">Cash or Margin</param>
+        public TDAmeritradeBrokerageModel(AccountType accountType = AccountType.Margin) : base(accountType)
+        {
+
+        }
+
+        /// <summary>
+        /// Returns true if the brokerage could accept this order. This takes into account
+        /// order type, security type, and order size limits.
+        /// </summary>
+        /// <remarks>
+        /// For example, a brokerage may have no connectivity at certain times, or an order rate/size limit
+        /// </remarks>
+        /// <param name="security">The security of the order</param>
+        /// <param name="order">The order to be processed</param>
+        /// <param name="message">If this function returns false, a brokerage message detailing why the order may not be submitted</param>
+        /// <returns>True if the brokerage could process the order, false otherwise</returns>
+        public override bool CanSubmitOrder(Security security, Order order, out BrokerageMessageEvent message)
+        {
+            if (!IsValidOrderSize(security, order.Quantity, out message))
+            {
+                return false;
+            }
+
+            message = null;
+            if (!new[] { SecurityType.Equity, SecurityType.Option }.Contains(security.Type))
+            {
+                message = new BrokerageMessageEvent(BrokerageMessageType.Warning, "NotSupported",
+                    StringExtensions.Invariant($"The {nameof(TDAmeritradeBrokerageModel)} does not support {security.Type} security type.")
+                );
+
+                return false;
+            }
+
+            if (!new[] { OrderType.Market, OrderType.Limit, OrderType.StopMarket, OrderType.StopLimit }.Contains(order.Type))
```

Comment:
> new [] {}

nit - Think should avoid creating these array on each verification. We can create some private field arrays holding there